### DISCUSSION TITLE
PNP search box not showing results

### DISCUSSION THREAD

**Discussion by Neenz13** [microsoft-search/pnp-modern-search#2826] [Q&A]
Hi there

I have a pnp search box where 2 search results are linked to. The first image is the first search results which is linked to the search box. Second image is the second search results which is linked to search box and the first search results. It uses refinablestring08 data from first search result so it is linked to supported documents and appears up on the second search results. What i am having issues with is the search box. When searching for certain files it is bringing up random files even when i set it so it goes by title. Even writing the exact title name of a file it still comes up with random files and the file i am looking for is either at the bottom or 2-3rd page. Its like its bringing up anything that starts with what i have searched for. Is there a query that i can use to sort this. I have tried to refine the searches by using properties from search results but to no avail. Thanks

![image](https://user-images.githubusercontent.com/126016838/223236551-9a84a3bd-2158-4ca5-bb95-ce0b8ef0410c.png)

![image](https://user-images.githubusercontent.com/126016838/223236589-409f5172-2471-4248-8b99-3a68e47f8b9d.png)

![image](https://user-images.githubusercontent.com/126016838/223243492-51ce516e-88f8-4266-a96f-e00c4d5b8f62.png)

![image](https://user-images.githubusercontent.com/126016838/223242933-53f498aa-ff96-4a49-8270-6680fd293e65.png)

**Comment by kasperbolarsen** [comment]
Hi , first of all, which version are you using?
Secondly you will have to let us know the Data source, the Query template and Refinement filters, otherwise we would be guessing

**Reply by Neenz13** [reply]
Apologies, using v4.8.0. Data source is a sharepoint document library where all the files are stored and where the search results are getting its data from. RefinableString101 is being used for document type and refinablestring08 is the document association used to link the first search results to the second search results.

![image](https://user-images.githubusercontent.com/126016838/223255848-3961f550-da77-46f8-b08f-1cf78d48c5a2.png)

![image](https://user-images.githubusercontent.com/126016838/223256027-f5a2e767-9bde-4514-9323-181a94a8003d.png)

![image](https://user-images.githubusercontent.com/126016838/223257991-947fde6d-c6e6-4fab-8cd3-885de82c5464.png)



Search box currently only has the below settings
![image](https://user-images.githubusercontent.com/126016838/223257397-110574ab-186f-4e7a-9018-0882698bf8f9.png)

**Reply by kasperbolarsen** [reply]
Ok, let me see if I understand the setup. You are using the search text from the search box in the search template on search results1. Search results2 is linked to search results1 using the Refinablestring08 value on the selected item in search results1 as the query.  Is this correct?

**Comment by Neenz13** [comment]
Have tried the below query to see if i can refine the searches by title and to pick up the policy but still bringing up files completely unrelated. Just wondering if the sort settings that i have in place will be a reason as to why search is not working properly

({searchTerms}
Path:https://xxx.sharepoint.com/sites/Policy-Hub/Policies
{|RefinableString101:"Policy"}) OR
(
(RefinableString03:{searchTerms} OR RefinableString101:{searchTerms})
Path:https://xxx.sharepoint.com/sites/Policy-Hub/Policies
{|RefinableString101:"Policy"})

**Reply by Neenz13** [reply]
Turns out that sorting is the reason. Sorting seems to override the query and caused the search to sort the searches instead of searching using the query (Policy). 

Have tried to tweak the sort but does not like this. Any suggestions?

**Reply by wobba** [reply]
What is the search box issue? Your screenshot shows the search result web part.

**Reply by Neenz13** [reply]
Hi there,

Searches are not appearing up correctly. As shown in the screenshot on the main comment when searching for specific items e.g below, i do not get the option at the top when searching. The search box seems to follow the sorting on the search result web part instead of bringing up the file/search inputted 


![image](https://user-images.githubusercontent.com/126016838/224565320-906b4f78-caae-4c72-b525-04dc85231af5.png)

**Reply by wobba** [reply]
You need to better explain what is wrong/not working.What do you mean "option at the top"?

**Reply by Neenz13** [reply]
Hi wobba, the issue is with the search box, it is not returning the correct files when i search. As the example above when i search for OP 5.6 it is returning files OP 2.1 first etc, and OP 5.6 is either showing on the next page or at the bottom of the first page. When i remove the "sorting" of files from the search results web part, the search box returns the correct file at the top. Its when i have sorting on the search box does not work correctly

**Reply by wobba** [reply]
@Neenz13 a search box does not return results and the search box does not have sorting capabilities. It's the result web parts which show results and have sorting. Is the issue when you type a keyword, and then execute the query that sort sorting is not applied?

If sorting is set correctly on the JSON request (check network tab), and results are not ordered like you expect, then it's an API issue, not an issue with the web parts. You can also try API queries via e.g. the SharePoint Search Query Tool to replicate. To get to the bottom of your issue you have to look at the search requests, and compare with what comes back.
